From foo@baz Wed May 12 04:07:32 PM CEST 2021
From: Quentin Perret <qperret@google.com>
Date: Wed, 12 May 2021 12:30:38 +0000
Subject: Revert "fdt: Properly handle "no-map" field in the memory region"
To: stable@vger.kernel.org
Cc: alexandre.torgue@foss.st.com, robh+dt@kernel.org, f.fainelli@gmail.com, ardb@kernel.org, gregkh@linuxfoundation.org, sashal@kernel.org
Message-ID: <20210512123038.3287684-2-qperret@google.com>

From: Quentin Perret <qperret@google.com>

This reverts commit 86ac82a7c708acf4738c396228be7b8fdaae4d99.
It is not really a fix, and the backport misses dependencies, which
breaks existing platforms.

Reported-by: Alexandre TORGUE <alexandre.torgue@foss.st.com>
Signed-off-by: Quentin Perret <qperret@google.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
 drivers/of/fdt.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/of/fdt.c
+++ b/drivers/of/fdt.c
@@ -1159,7 +1159,7 @@ int __init __weak early_init_dt_reserve_
 					phys_addr_t size, bool nomap)
 {
 	if (nomap)
-		return memblock_mark_nomap(base, size);
+		return memblock_remove(base, size);
 	return memblock_reserve(base, size);
 }
 
